src/foam/core/stdlib.js

```diff
@@ -796,8 +793,9 @@ foam.LIB({
         // we will always use the comparator of higher precedence.
         var aType = typeOf(a);
         var bType = typeOf(b);
-        return (aType.ordinal >= bType.ordinal) ? aType.compare(a, b) :
-            this.comparatorNegate(bType.compare(b, a));
+        return aType.ordinal > bType.ordinal ? 1 : aType.ordinal < bType.ordinal ? -1 : aType.compare(a, b);
```

Comment:
I believe some of the other comparators have a check that is performed (e.g. `if ( ! foam.<type>.isInstance(b) ) return 1`)
The only case that I can think of that depends on those checks now is `return -foam.typeOf(other).compare(other, this);` (the change you made in FObject.js).

I believe those can be removed if FObject utilize `foam.util.compare` instead of the comparators directly